File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>
using namespace std;

vector<string> solution(vector<string> players, vector<string> callings) {
    vector<string> answer;
    map<string,int> s;
    for(int i=0;i<players.size();i++){
        s[players[i]]=i;
    }
    for(int i=0;i<callings.size();i++){
        string a = callings[i];
        int b = s[a];
        string temp = players[b-1];
        players[b-1] = a;
        players[b] = temp;
        s[temp]++;
        s[a]--;
    }
    answer = players;
    return answer;
}

File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.js
function solution(players, callings) {
    var answer = [];
    let run = new Map();
    for(let i=0;i<players.length;i++){
        run.set(players[i],i);
    }
    for(let i=0;i<callings.length;i++){
        let now = callings[i];
        let level = run.get(now);
        run.set(now,level-1);
        let front = players[level-1];
        run.set(front,run.get(front)+1);
        players[level-1] = now;
        players[level] = front;
    }
    return players;
}